pallets/storage-bench/src/lib.rs

```diff
@@ -0,0 +1,390 @@
+            T::UpdateOrigin::ensure_origin(origin)?;
+            let mut receivers = Vec::new();
+            let mut iter = if utxo_idx == 0 {
+                ShardsTwoxIdentity::<T>::iter_prefix(shard_idx)
+            } else {
+                let raw_key = ShardsTwoxIdentity::<T>::hashed_key_for(
+                    shard_idx,
+                    Self::encode(utxo_idx as u64 - 1),
+                );
+                ShardsTwoxIdentity::<T>::iter_prefix_from(shard_idx, raw_key)
+            };
+            for _ in 0..amount {
+                match iter.next() {
+                    Some((_, next)) => receivers.push(next),
+                    _ => return Err(Error::<T>::NullStorageItem.into()),
+                }
+            }
+            Self::deposit_event(Event::ShardElementBatchRead(amount));
+            Ok(())
+        }
+    }
+
+    impl<T: Config> Pallet<T> {
+        fn encode(n: u64) -> u64 {
+            let mut bytes = n.to_ne_bytes();
+            bytes.reverse();
+            u64::from_ne_bytes(bytes)
+        }
+
+        pub(crate) fn do_insert_shard_element_identity_identity(
+            shard_idx: u8,
+            utxo: Utxo,
+            encrypted_note: EncryptedNote,
+            lex_encoding: bool,
+        ) -> DispatchResult {
+            let utxo_idx = ShardsIdentityIdentityIndices::<T>::get(shard_idx);
+            let encoded_utxo_idx = if lex_encoding {
+                Self::encode(utxo_idx)
+            } else {
+                utxo_idx
+            };
+            ShardsIdentityIdentity::<T>::insert(
+                shard_idx,
+                encoded_utxo_idx,
+                (utxo, encrypted_note),
+            );
+            ShardsIdentityIdentityIndices::<T>::insert(shard_idx, utxo_idx + 1);
+            Ok(())
+        }
+
+        pub(crate) fn do_insert_shard_element_twox_identity(
+            shard_idx: u8,
+            utxo: Utxo,
+            encrypted_note: EncryptedNote,
+            lex_encoding: bool,
+        ) -> DispatchResult {
+            let utxo_idx = ShardsTwoxIdentityIndices::<T>::get(shard_idx);
+            let encoded_utxo_idx = if lex_encoding {
+                Self::encode(utxo_idx)
+            } else {
+                utxo_idx
+            };
+            ShardsTwoxIdentity::<T>::insert(
+                shard_idx,
+                encoded_utxo_idx,
+                (utxo, encrypted_note),
+            );
+            ShardsTwoxIdentityIndices::<T>::insert(shard_idx, utxo_idx + 1);
+            Ok(())
+        }
+
+        pub(crate) fn do_insert_shard_element_twox_twox(
+            shard_idx: u8,
+            utxo: Utxo,
+            encrypted_note: EncryptedNote,
+        ) -> DispatchResult {
+            let utxo_idx = ShardsTwoxTwoxIndices::<T>::get(shard_idx);
+            ShardsTwoxTwox::<T>::insert(shard_idx, utxo_idx, (utxo, encrypted_note.clone()));
+            ShardsTwoxIdentityIndices::<T>::insert(shard_idx, utxo_idx + 1);
```

Comment:
why do you insert twoxIdentity in the twoxtwox insert fn?
could this be left over from copy-pasting?
seems this would slow down the twoxtwox results needlessly